src/main/java/lotto/domain/LottoNumbers.java

```diff
@@ -3,37 +3,70 @@
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.Collections;
+import java.util.HashSet;
 import java.util.List;
+import java.util.Set;
 import java.util.stream.Collectors;
 
 public class LottoNumbers {
-	private static final int MATCHED_NUMBER_PLUS = 1;
-	private static final int NON_MATCHED_NUMBER = 0;
+	private static final String ERROR_DUPLICATE_LOTTO_NUMBER = "[ERROR] 로또 번호는 중복 될 수 없습니다.";
+	private static final String ERROR_LOTTO_NUMBER_LENGTH = "[ERROR] 로또 번호는 6자리 숫자를 가져야 합니다.";
 	private static final String DELIMITER = ", ";
-	private final List<Integer> lottoNumbers;
+	public static final int MATCHED_NUMBER = 1;
+	private static final int NON_MATCHED_NUMBER = 0;
+	public static final int MIN_RANGE_VALUE = 0;
+	public static final int MAX_RANGE_VALUE = 6;
+	private final List<LottoNumber> lottoNumbers;
 
-	public LottoNumbers(int... lottoNumbers) {
+	public LottoNumbers(final int... lottoNumbers) {
 		this(Arrays.stream(lottoNumbers)
-			.boxed()
+			.mapToObj(LottoNumber::new)
 			.collect(Collectors.toList()));
 	}
 
-	public LottoNumbers(List<Integer> lottoNumbers) {
-		this.lottoNumbers = Collections.unmodifiableList(new ArrayList<>(lottoNumbers));
+	public LottoNumbers(final String lottoNumbers) {
+		this(validateStringLottoNumbers(lottoNumbers));
+	}
+
+	public LottoNumbers(final List<LottoNumber> lottoNumbers) {
+		validateLottoNumberLength(lottoNumbers);
+		this.lottoNumbers = new ArrayList<>(lottoNumbers);
 	}
 
-	public int isMatch(int winningNumber) {
+	public int ifMatchCount(LottoNumber winningNumber) {
 		if (lottoNumbers.contains(winningNumber)) {
-			return MATCHED_NUMBER_PLUS;
+			return MATCHED_NUMBER;
 		}
 		return NON_MATCHED_NUMBER;
 	}
 
-	public String StringValues() {
-		String values = "";
-		for (Integer lottoNumber : lottoNumbers) {
-			values += String.valueOf(lottoNumber) + DELIMITER;
+	public List<LottoNumber>  getLottoNumbers() {
+		return Collections.unmodifiableList(lottoNumbers);
+	}
+
+	private static List<LottoNumber> validateStringLottoNumbers(String lottoNumbers) {
+		String[] stringNumberArray = lottoNumbers.split(DELIMITER);
+		validateDuplicate(stringNumberArray);
+		return Arrays.stream(stringNumberArray)
+			.map(LottoNumber::new)
+			.collect(Collectors.toList());
+	}
+
+	private static void validateDuplicate(String[] stringNumberArray) {
+		Set<String> duplicateCheck = new HashSet<>();
+		for (String number : stringNumberArray) {
+			duplicateCheck.add(number);
+		}
```

Comment:
Collections API를 사용해보는 건 어떨까요?

```suggestion
		Collections.addAll(duplicateCheck, stringNumberArray);
```